File: java/src/main/java/com/teamlab/kotlin/dagger/issue/MainFragment.java
package com.teamlab.kotlin.dagger.issue;

import android.app.Fragment;
import android.os.Bundle;
import android.util.Log;

import javax.inject.Inject;

public class MainFragment extends Fragment {

    @Inject
    String string;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ((MainActivity) getActivity()).getComponent().inject(this);
        Log.v("MainFragment", string);
    }
}

File: kotlin/src/main/kotlin/com/teamlab/kotlin/dagger/issue/MainFragment.kt
package com.teamlab.kotlin.dagger.issue

import android.app.Fragment
import android.os.Bundle
import android.util.Log

import javax.inject.Inject

class MainFragment : Fragment() {

    @Inject
    lateinit var string: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        (activity as MainActivity).component.inject(this)
        Log.v("MainFragment", string)
    }
}
